base/mac/files/file_extended_attribute.h

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BASE_MAC_FILES_FILE_EXTENDED_ATTRIBUTE_H_
+#define BRAVE_BASE_MAC_FILES_FILE_EXTENDED_ATTRIBUTE_H_
+
+#include <vector>
+
+#include "base/files/file_path.h"
+
+namespace brave {
+
+// Extracts the value of an extended file attribute specified by |name| from the
+// file specified by the |path|. The result is placed in the |value|. On
+// success returns 0. Otherwise, returns ERRNO value. Error values of
+// interest:
+// ENOATTR - the attribute with the given |name| was not found.
+// ENOTSUP - the file system doesn't support (or disabled) extended attributes.
+// ENODATA - the value could not be retrieved.
+int GetFileExtendedAttribute(const base::FilePath& path,
```

Comment:
I don't really see any reason to expose this stuff in base, I would just keep it all in the referrals service

Reply:
It seemed like a more generic functionality that might be useful elsewhere at some point. Also, makes it more awkward to unit test - I'd have to expose it in the referrals service header.

Reply:
Moved these files under `components/brave_referrals/browser/` in https://github.com/brave/brave-core/pull/20262/commits/83030888a121b765276ee249336f92af0561e05f